src/model/consent/consent.ts

```diff
@@ -0,0 +1,113 @@
+/* istanbul ignore file */
+
+/*
+ * This flag is to ignore BDD testing for model
+ * which will be addressed in a future ticket
+ */
+
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the 'License') and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Raman Mangla <ramanmangla@google.com>
+ --------------
+ ******/
+
+import Knex from 'knex'
+
+/*
+ * Interface for Consent resource type
+ */
+export interface Consent {
+  id: string;
+  initiatorId: string;
+  participantId: string;
```

Comment:
Are we sure these are required? Won't be want to do a lookup by ID and need these to be empty during that lookup or no?

Reply:
The fields are `required` to enforce that they are a part of each request that the auth-service receives (which is the case in the sequence diagrams). The `retrieve` consent model just requires a `consentID` (String) for lookup and doesn't concern the interface. Do you feel that's justified?

Reply:
Sounds like the DFSP is responsible for specifying the Consent ID in the POST (create) request. So this should be required for sure.